bin/cli.js

```diff
@@ -147,6 +154,35 @@ Example
       runTransform(target, command, program, { component, prop })
     );
 
+  program
```

Comment:
This failed to transform the following code 

```jsx
<Button renderRouterLink={(props) => <Link {...props} />} to="/protected">
      This link is protected
</Button>
```

`npx carbon-codemod replace-prop-value . carbon-react/lib/components/button to "/protected" "/private"`

```
npx carbon-codemod replace-prop-value . carbon-react/lib/components/button to "/protected" "/private"  
jscodeshift --verbose=2 --ignore-pattern=**/node_modules/** --transform /Users/jamie/carbon-codemod/transforms/replace-prop-value/replace-prop-value.js /Users/jamie/app-1 --component=carbon-react/lib/components/button --prop=to --oldValue=/protected --newValue=/private
```

I thought it might be the leading `/` but removing it had no effect.

Reply:
This now runs, but the replacement is ends up being wrapped in a `JSXExpressionContainer` instead of being a string.

```diff
-- <Button renderRouterLink={(props) => <Link {...props} />} to="/protected">
++ <Button renderRouterLink={(props) => <Link {...props} />} to={"/private"}>
```

However, there might be some cases where we need the container, e.g. when the user is trying to set the prop to be a number. We potentially have the same problem for the existing value, users might be trying to match only strings, or only numbers.

We'll need a way to determine this, we can either use an optional argument
e.g. 

```
npx carbon-codemod replace-prop-value ... 1 2 --replaceWithContainer
```

or we can use a mix of `parseInt/parseFloat` to work out if the argument is numeric, if it is we will treat it as a number and put it in a container. Users could then escape this by passing escaped quotes. e.g.

```
npx carbon-codemod replace-prop-value ... \"1\" \"2\"
```

We'd have to handle single and double quotes.

Although the second solution sounds messy, I think it's probably the better solution.

Commander has some options which allows us to coerce the value before it's passed to the codemod which I think will make things simpler. In the codemod we could then rely on the `typeof/isNaN`

https://github.com/tj/commander.js/#custom-option-processing

Reply:
My idea for this was put everything in a container and it'll always work correctly, and the formatting for strings should be handled by formatters in consuming apps. It won't always be but at least a string inside a container is a valid prop

If an app is running prettier on commit for example, it would just remove the brackets for us here

Reply:
I don't actually see what you mean about the same problem for the existing value? I'll message you on slack about it